DeepSpeech.py

```diff
@@ -1170,17 +1174,17 @@ def _next_epoch(self):
         if (FLAGS.early_stop is True) and (FLAGS.validation_step > 0) and (len(self._dev_losses) >= FLAGS.earlystop_nsteps):
 
             # Calculate the mean of losses for past epochs
-            mean_loss = np.mean(self._dev_losses[-FLAGS.earlystop_nsteps:-2])
+            mean_loss = np.mean(self._dev_losses[-FLAGS.earlystop_nsteps:-1])
             # Calculate the standard deviation for losses from validation part in the past epochs
-            std_loss = np.std(self._dev_losses[-FLAGS.earlystop_nsteps:-2])
+            std_loss = np.std(self._dev_losses[-FLAGS.earlystop_nsteps:-1])
             # Update the list of losses incurred
             self._dev_losses = self._dev_losses[-FLAGS.earlystop_nsteps:]
             log_debug('Current validation loss: %f, std_of_loss(n_step) :%f mean_loss(n_step): %f' % (self._dev_losses[-1], std_loss, mean_loss))
 
             # Making sure slight fluctuations don't bother the early stopping from performing
-            if self._dev_losses[-1] > np.max(self._dev_losses[:-2]) or (abs(self._dev_losses[-1] - mean_loss) < FLAGS.estop_mean_thresh and std_loss < FLAGS.estop_std_thresh):
+            if self._dev_losses[-1] > np.max(self._dev_losses[:-1]) or (abs(self._dev_losses[-1] - mean_loss) < FLAGS.estop_mean_thresh and std_loss < FLAGS.estop_std_thresh):
                 # Time to early stop
-                log_info('Early stop triggered!! Validation_loss:%f std_of_loss(n_step) :%f mean_loss(n_step): %f' % ((self._dev_losses[-1], std_loss, mean_loss)))
+                log_info('Early stop triggered, validation_loss: %f std_of_loss(n_step): %f mean_loss(n_step): %f' % ((self._dev_losses[-1], std_loss, mean_loss)))
```

Comment:
Here you could reason the early stop by having two different messages for the two `or` parts of this `if` block (by making it an `elif`).
E.g. "Early stop, as ..."
Please don't use underscores and abbreviations.